src/main/java/org/apache/commons/lang3/CachedRandomBits.java

```diff
@@ -52,6 +52,14 @@ final class CachedRandomBits {
      */
     private int bitIndex;
 
+    /**
+     * The maximum size of the cache.
+     *
+     * <p>
+     * This is dictated by the {@code if (bitIndex >> 3 >= cache.length)} in the {@link #nextBits(int)} method.
```

Comment:
If the `3` in this expression MUST match the `3` in the expression building `cacheSize` in the `random(...)` method, then it should be refactored from a magic number in to a constant IMO.